### PR TITLE
Automated Delayed replication - Failback scenario 

### PR DESCRIPTION
**Pull request by VasishtaShastry**
Added automation for delay scenario with failback to existing handler which
manages rbd mirror delayed replication scenarios.

Wrokflow of Tc:
polarion-id: CEPH-11493 - Delayed Replication-Failback with delay set.

    Set delay at primary as image config
    Failback the images
    Record md5sum value of images.
    Set delay at primary as image config
    Perform some IO operations on primary
    observe image data till mentioned delay that
        there are no mirroring activity between clusters.

File changes:
modified:   suites/quincy/rbd/tier-2_rbd_upgrade_5x_to_6x.yaml
Added configs for TC automated as part of this commit

modified:   tests/rbd_mirror/rbd_mirror_utils.py
1. Added config ignore_command_failure to wait_for_status
This is required at the time of images resyncs when we expect command failures like
image unavailable during earlier wait periods.
2. Added capability to wait_for_status method to wait for specific value of description.
3. Removed a redundant code comment.
4. Added modules fail_back and prepare_for_failback to handle failback for journal based
   mirroring scenarios.

modified: tests/rbd_mirror/test_delayed_replication_handler.py
Enhanced handler to accomodate new testcase and also some minor optimization.

# Description

Please include Automation development guidelines. Source of Test case - New Feature/Regression Test/Close loop of customer BZs
<details>

<summary>click to expand checklist</summary>

- [ ] Create a test case in Polarion reviewed and approved.
- [ ] Create a design/automation approach doc. Optional for tests with similar tests already automated.
- [ ] Review the automation design
- [ ] Implement the test script and perform test runs
- [ ] Submit PR for code review and approve
- [ ] Update Polarion Test with Automation script details and update automation fields
- [ ] If automation is part of Close loop, update BZ flag qe-test_coverage “+” and link Polarion test
</details>

### CODE REVIEW

**Comment by rkhambad** [comment]
Please solve the tox issue and make PR ready for review.

File: suites/quincy/rbd/tier-2_rbd_upgrade_5x_to_6x.yaml [hunk 1]
```diff
@@ -9,192 +9,6 @@
 #===============================================================================================
 
 tests:
-  - test:
```
Comment (rkhambad) [inline suites/quincy/rbd/tier-2_rbd_upgrade_5x_to_6x.yaml:12]: un-comment this part of test suite.

**Comment by rkhambad** [comment]
@VasishtaShastry  is it open for review or you want to make any more changes in the PR.

File: suites/quincy/rbd/tier-2_rbd_upgrade_5x_to_6x.yaml (hunk 1 above)
Comment (VasishtaShastry) [reply suites/quincy/rbd/tier-2_rbd_upgrade_5x_to_6x.yaml]: Done.

**Comment by VasishtaShastry** [comment]
Log link http://magna002.ceph.redhat.com/cephci-jenkins/cephci-run-M6NO6D

File: tests/rbd_mirror/rbd_mirror_utils.py [hunk 2]
```diff
@@ -1211,3 +1242,69 @@ def parallel_bench_on_n_images(
...
+    if not orderly:
+        if demoted.mirror_status("image", imagespec, "state") != "up+stopped":
+            log.error(
+                "Aborting failback, seems like non-orderly failover was not performed"
+            )
+            return 1
+
+        prepare_for_failback(demoted, imagespec)
+
+    else:
```
Comment (sunilkumarn417) [inline tests/rbd_mirror/rbd_mirror_utils.py:1294]: use elif

Comment (VasishtaShastry) [reply tests/rbd_mirror/rbd_mirror_utils.py]: Done.

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**